spec.md

```diff
@@ -179,6 +220,47 @@ physically present in the final circuit. Refer to the description of the
 instance statement for details on how to instantiate a module
 ([@sec:instances]).
 
+### Public Modules
+
+A public module is a module that is marked with the `public`{.firrtl} keyword.
+A public module may be instantiated outside by other circuits.
+
+The example below shows a public module `Foo`:
+
+``` firrtl
+public module Foo:
+  input a: UInt<1>
+  output b: UInt<1>
+
+  connect b, a
+```
+
+A public module has a number of restrictions:
+
+1. A public module may have no ports of uninferred width.
+1. A public module may have no ports of abstract reset type.
+1. A public module may have no ports of input probe type.
+1. A `RWProbe`{.firrtl} may not be used to access a public module's ports.
+1. A public module's ports are lowered using one of the port lowering ABIs
+   defined in the FIRRTL ABI specification.
+1. A public module may be instantiated by other modules within a circuit, but
+   the behavior of the module must not be affected by these instantiations.
+
+The last point requires clarification.  A public module may or may not be
```

Comment:
Let's not use the phrase "may or may not" here, since either one alone sounds like a prescription.

Reading this closer, this whole paragraph is confusing to me. What does it mean to be "compiled" with respect to the rest of the spec? It comes off as if it can be instantiated without being instantiated :)

Reply:
The problem here is that there is no good way of indicating commentary in the spec. The RISC-V specifications have shaded blocks where they do this and this may be useful to add to the FIRRTL spec. 

None of the body of this paragraph has any bearing on the spec and is only providing commentary on how to interpret the last item (line 246) if you are building a FIRRTL compiler. If you need to produce Verilog from a public module, this will make sense to you. If you are trying to understand FIRRTL, then it has no meaning.

I'll move some of this language into the ABI doc as that is more appropriate.